src/Components/Components/src/Routing/Router.cs

```diff
@@ -190,28 +192,31 @@ private void Refresh(bool isNavigationIntercepted)
             }
         }
 
-        private async Task RunOnNavigateAsync(string path)
+        private async ValueTask<bool> RunOnNavigateAsync(string path, Task previousOnNavigate)
         {
             // If this router instance does not provide an OnNavigateAsync parameter
             // then we render the component associated with the route as per usual.
             if (!OnNavigateAsync.HasDelegate)
             {
-                return;
+                return true;
             }
 
             // If we've already invoked a task and stored its CTS, then
-            // cancel the existing task.
-            _onNavigateCts?.Dispose();
+            // cancel that existing CTS.
+            _onNavigateCts?.Cancel();
+            // Then make sure that the task has been completed cancelled or
+            // completed before continuing with the execution of this current task.
+            await previousOnNavigate;
```

Comment:
I'm sure this `await previousOnNavigate` is for a good reason, but it seems subtle. I can see that due to the chain of events triggered by L206, the previous task will immediately become cancelled, so the await on L209 shouldn't have to really wait.

So it makes me wonder why we couldn't just ignore `previousOnNavigate`, since we know it will have been cancelled and that its continuation is going to no-op (not refresh the UI).

Also I spent a while being confused about error handling here, because I thought that if `previousOnNavigate` had failed or been cancelled, we'd be blocking all subsequent navigation because we're awaiting it. Then later I realised that `previousOnNavigate` can never fail or be cancelled because it's entirely managed within `RunOnNavigateWithRefreshAsync`, which only ever sets it to completed.

So, it would be good to clarify in the comment why we want to await this, and why we know it will never fail or be cancelled :)

Reply:
Update: in fact I realise now that `previousOnNavigate` will *not* actually have been cancelled; it will just have been completed due to the chain of events triggered by L206. So the existing comment is a bit misleading since it suggests it might be cancelled.

All these CTSs and TCSs are quite hard to keep track of! I do still wonder why we need to await `previousOnNavigate` here.

Reply:
I recommended this when reviewing the the CTS usage. I didn't want to have to think about the possible consequences of the meat of RunOnNavigateWithRefreshAsync running concurrently of itself, so in my mind this simplified things. I cannot point to any specific bug this prevents though.

Reply:
@halter73 Thanks for posting this response. I also updated the commenting around here in a follow-up PR to make ti clearer.